generator/js_generator.cc

```diff
@@ -2254,7 +2254,7 @@ void Generator::GenerateClassXid(const GeneratorOptions& options,
 
 void Generator::GenerateOneofCaseDefinition(
     const GeneratorOptions& options, io::Printer* printer,
-    const OneofDescriptor* oneof) const {
+    const OneofDescriptor* oneof, const Descriptor* desc) const {
```

Comment:
What is needed is probably changing className = GetMessagePath to `oneof->containing_type()->name())`. I don't think taking `desc` is necessary.